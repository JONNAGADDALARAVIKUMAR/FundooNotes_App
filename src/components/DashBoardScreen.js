import React, { Component } from 'react'
import {View, Text} from 'react-native'

export default class extends Component {
    render() {
        return (
            <View style = {{backgroundColor: '#dbb6c3'}}>
                <Text style = {{textAlignVertical: 'center', textAlign: 'center'}}>
                    Page Under Development
                </Text>
            </View>
        )
    }
}