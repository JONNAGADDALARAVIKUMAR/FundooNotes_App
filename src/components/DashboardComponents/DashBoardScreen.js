import React, { Component } from 'react';
import {View, ScrollView} from 'react-native';
import ToolBar from './ToolBar';
import BottomBar from './BottomBar';
import ViewNotes from './NotesView';
import {Snackbar, Provider, Portal, Modal} from 'react-native-paper';
import UserNoteServices from '../../../services/UserNoteServices';
import Profile from './Profile';
import UserServices from '../../../services/UserServices';

export default class Dashboard extends Component {
    constructor(props){
        super(props)
        this.state = {
            listView: false,
            showEmptyNoteSnackbar : false,
            showDeletedNoteSnackbar : false,
            showProfile: false,
            photoURL: ''
        }
    }

    componentDidMount = async () => {
        this.getProfileImage();
        if(this.props.route.params != undefined) {
            if(this.props.route.params.isEmptyNote != undefined) {
                await this.setState({
                    showEmptyNoteSnackbar : true
                })
            }
            if(this.props.route.params.isNoteDeleted != undefined) {
                await this.setState({
                    showDeletedNoteSnackbar : true
                })
            }
        }
    }

    getProfileImage = async () => {
        await UserServices.getDetails()
        .then(async details => {
            if(details.imageURL != undefined) {
                await this.setState({
                    photoURL : details.imageURL
                })
            } else {
                await this.setState({
                    photoURL : ''
                })
            }
        })
        .catch(async error => {
            if(error.code == 'storage/object-not-found') {
                await this.setState({
                    photoURL : ''
                })
            }
            console.log(error);
        })
    }

    selectView = () => {
        const {onPress} = this.props
        this.setState({
            listView: !this.state.listView
        })
        //onPress();
    }

    emptyNoteSnackbarHandler = async () => {
        const {onDismiss} = this.props
        await this.setState({ 
            showEmptyNoteSnackbar : false
        })
        this.props.navigation.setParams({isEmptyNote : false})
        //onDismiss()
    }

    deletedNoteSnackbarHandler = async () => {
        const {onDismiss} = this.props
        await this.setState({ 
            showDeletedNoteSnackbar : false
        })
        this.props.navigation.setParams({isNoteDeleted : false})
        //onDismiss()
    }

    restoreNotes = () => {
        const {onPress} = this.props
        UserNoteServices.restoreNoteInFirebase(this.props.route.params.title, this.props.route.params.note, this.props.route.params.noteKey, this.props.route.params.archivedStatus, this.props.route.params.labelNoteKeys)
            .then(() => this.props.navigation.push('Home', {screen : 'Notes'}))
            .catch(error => console.log(error))
        //onPress()
    }

    handleProfile = async () => {
        await this.setState({
            showProfile: !this.state.showProfile
        })
        this.getProfileImage();
    }

    render() {
        return (
            <Provider>
            <View style = {{backgroundColor: '#f2d5e5', height: '100%'}}>
                <ToolBar 
                    navigation = {this.props.navigation} 
                    onPress = {this.selectView} 
                    listView = {this.state.listView}
                    handleProfile = {this.handleProfile}
                    photoURL = {this.state.photoURL}/>

                <ScrollView>   
                    <ViewNotes 
                        navigation = {this.props.navigation} 
                        changeLayout = {this.state.listView} 
                        deletedStatus = {false}
                        archivedStatus = {false}/>
                </ScrollView>

                <BottomBar 
                    navigation = {this.props.navigation}/>

                <Snackbar
                    style = {{marginBottom : 80}}
                    visible={this.state.showEmptyNoteSnackbar}
                    onDismiss={this.emptyNoteSnackbarHandler}
                    duration = {3000}>
                    Empty Note Discarded
                </Snackbar>
                <Snackbar
                    style = {{marginBottom : 100}}
                    visible = {this.state.showDeletedNoteSnackbar}
                    onDismiss = {this.deletedNoteSnackbarHandler}
                    duration = {5000}
                    action = {{
                        label : 'Undo',
                        onPress : this.restoreNotes
                    }}>
                    Note Moved to Bin
                </Snackbar>
                    <Portal>
                        <Modal
                            visible = {this.state.showProfile}>
                            <Profile 
                                handleProfile = {this.handleProfile} 
                                navigation = {this.props.navigation}
                                photoURL = {this.state.photoURL}/>
                        </Modal>
                    </Portal>
            </View>
            </Provider>
        )
    }
}