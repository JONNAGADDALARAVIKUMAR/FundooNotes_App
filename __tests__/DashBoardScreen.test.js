import React from 'react';
import {shallow} from 'enzyme';
import DashBoardScreen from '../src/components/DashboardComponents/DashBoardScreen';

describe('test DashBoard Screen', () => {
    it('test when render should match to snapshot', async () => {
        const component = shallow(<DashBoardScreen/>)
        expect(component).toMatchSnapshot();
    })

    it('On pressing Logout Button navigateToLogInScreen method should Navigate to LogIn Screen', async () => {
        const navigation = { navigate : jest.fn() }
        const onPressEvent = jest.fn();
        const component = shallow(<DashBoardScreen onPress = {onPressEvent} navigation = {navigation}/>)
        const instance = component.instance();
        
        await instance.navigateToLogInScreen();
        expect(onPressEvent).toHaveBeenCalled();
        expect(navigation.navigate).toBeCalledWith("LogIn");
    })
})