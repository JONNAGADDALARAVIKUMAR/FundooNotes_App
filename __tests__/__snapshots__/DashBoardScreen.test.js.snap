// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DashBoard Screen test when render should match to snapshot 1`] = `ShallowWrapper {}`;
